File: solutions/ruby/resistor-color/2/resistor_color.rb
module ResistorColor

  def self.colors
    %w[black brown red orange yellow green blue violet grey white]
  end

  COLORS = self.colors

  BAND = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  def self.color_code(color)
    BAND[color.to_sym]
  end

end

File: solutions/elixir/resistor-color/2/lib/resistor_color.ex
defmodule ResistorColor do

  @colors_with_digits Enum.with_index([
    :black,
    :brown,
    :red,
    :orange,
    :yellow,
    :green,
    :blue,
    :violet,
    :grey,
    :white
  ])

  # Generate a function clause for each color
  for {color, digit} <- @colors_with_digits do
    def code(unquote(color)) do
      unquote(digit)
    end
  end
  
end
